bin/improver-weighted-blending

```diff
@@ -283,6 +287,24 @@ def main():
     else:
         coord_unit = 'no_unit'
 
+    # For blending across models, only blending across "model_id" is directly
+    # supported. This is because the blending coordinate must be sortable, in
+    # order to ensure that the data cube and the weights cube have coordinates
+    # in the same order for blending. Whilst the model_configuration is
+    # sortable itself, as it is associated with model_id, which is the
+    # dimension coordinate, sorting the model_configuration coordinate can
+    # result in the model_id coordinate becoming non-monotonic. As dimension
+    # coordinates must be monotonic, this leads to the model_id coordinate
+    # being demoted to an auxiliary coordinate. Therefore, for simplicity
+    # model_id is used as the blending coordinate, instead of
+    # model_configuration.
+    # TODO: Support model_configuration as a blending coordinate directly.
+    if args.coordinate == "model_configuration":
+        args.coordinate = "model_id"
+        dict_coord = "model_configuration"
+    else:
+        dict_coord = args.coordinate
```

Comment:
Could this be done inside the "calculate_blending_weights" function please.  "dict_coord" is only needed for calculating the weights, and it reduces the number of arguments that need to be passed in.

Reply:
I could do that but, although you're correct that `dict_coord` is only needed for calculating weights, this section of code is also resetting `args.coordinate`, which is passed into the `SpatialWeightsPlugin` and the `WeightedBlendAcrossBlendDimension` plugins. Therefore if I moved this section of code into `calculate_blending_weights`, I'd have to pass the `blend_coord` out of the `calculate_blending_weights` function. What do you think?

Reply:
Ah OK, I missed that you were modifying args.coordinate.  In which case I think you're right this loop should be outside the function, but please could you have a new variable ("blend_coord") rather than modifying args.coordinate.

Reply:
Added new variable ("blend_coord"), rather than modifying args.coordinate.